src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/InputIllegalIdentifierNameUnderscore.java

```diff
@@ -1,3 +1,4 @@
+//non-compiled with javac: Compilable with Java8
```

Comment:
The underscore (`_`) is not a valid name for a variable since java9.

Reply:
@pbludov Are you saying you can't name variables like `MY_EXAMPLE_HERE` anymore for constants?

Reply:
no, just the underscore itself: `_` invalid, but `__` valid. Comes from http://openjdk.java.net/jeps/302